jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/internal/memory/G1HeapSummaryHandler.java

```diff
@@ -0,0 +1,121 @@
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.EDEN_SIZE_DELTA;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.EDEN_USED;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.REGION_COUNT;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.SURVIVOR_SIZE;
+import static io.opentelemetry.contrib.jfr.metrics.internal.RecordedEventHandler.defaultMeter;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleHistogram;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.contrib.jfr.metrics.internal.RecordedEventHandler;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+import jdk.jfr.consumer.RecordedEvent;
+
+/**
+ * This class handles G1HeapSummary JFR events. For GC purposes they come in pairs. Basic heap
+ * values are sourced from GCHeapSummary - this is young generational details
+ */
+public final class G1HeapSummaryHandler implements RecordedEventHandler {
+  private static final Logger logger = Logger.getLogger(G1HeapSummaryHandler.class.getName());
+
+  private static final String METRIC_NAME_MEMORY = "process.runtime.jvm.memory.used";
+  private static final String METRIC_DESCRIPTION_MEMORY = "Heap utilization";
+  private static final String EVENT_NAME = "jdk.G1HeapSummary";
+  private static final String BEFORE = "Before GC";
+  private static final String AFTER = "After GC";
+  private static final String GC_ID = "gcId";
+  private static final String WHEN = "when";
+  private static final Attributes ATTR_MEMORY_EDEN_USED = Attributes.of(ATTR_USAGE, EDEN_USED);
+  private static final Attributes ATTR_MEMORY_EDEN_SIZE = Attributes.of(ATTR_USAGE, EDEN_SIZE);
+  private static final Attributes ATTR_MEMORY_EDEN_SIZE_DELTA =
+      Attributes.of(ATTR_USAGE, EDEN_SIZE_DELTA);
+  private static final Attributes ATTR_MEMORY_SURVIVOR_SIZE =
+      Attributes.of(ATTR_USAGE, SURVIVOR_SIZE);
+  private static final Attributes ATTR_MEMORY_REGIONS = Attributes.of(ATTR_USAGE, REGION_COUNT);
+
+  private final Map<Long, RecordedEvent> awaitingPairs = new HashMap<>();
+
+  private DoubleHistogram memoryHistogram;
+
+  public G1HeapSummaryHandler() {
+    initializeMeter(defaultMeter());
+  }
+
+  @Override
+  public void initializeMeter(Meter meter) {
+    memoryHistogram =
+        meter
+            .histogramBuilder(METRIC_NAME_MEMORY)
+            .setDescription(METRIC_DESCRIPTION_MEMORY)
+            .setUnit(BYTES)
+            .build();
+  }
+
+  @Override
+  public String getEventName() {
+    return EVENT_NAME;
+  }
+
+  @Override
+  public void accept(RecordedEvent ev) {
+    String when;
+    if (ev.hasField(WHEN)) {
+      when = ev.getString(WHEN);
+    } else {
+      logger.fine(String.format("G1 GC Event seen without when: %s", ev));
+      return;
+    }
+    if (!(when.equals(BEFORE) || when.equals(AFTER))) {
```

Comment:
Since we're not explicitly checking if `when != null`, we should probably reverse these conditions to be safe: 
```
if (!(BEFORE.equals(when) || AFTER.equals(when))) {
```

Reply:
Fixed.